Core/World/Entities/Players/Player.cs

```diff
@@ -912,6 +917,96 @@ private void ChangePlayerWeaponSlot(WeaponSlot slot)
         }
     }
 
+    private static int GetWeaponGroupIndex(TickCommands tickCommand) =>
+        (int)tickCommand - (int)TickCommands.WeaponGroup1;
+    
+    private static readonly TickCommands[] WeaponGroupCommands = new TickCommands[]
+    {
+        TickCommands.WeaponGroup1,
+        TickCommands.WeaponGroup2,
+        TickCommands.WeaponGroup3,
+        TickCommands.WeaponGroup4
+    };
+
+    private static TickCommands GetWeaponGroupCommand(TickCommand tickCommand)
+    {
+        for (int i = 0; i < WeaponGroupCommands.Length; i++) {
+            if (tickCommand.Has(WeaponGroupCommands[i]))
+                return WeaponGroupCommands[i];
+        }
+        return TickCommands.None;
+    }
+
+    private ConfigWeaponSlots[,] GetWeaponGroups() => new ConfigWeaponSlots[4,3] {
+        {
+            WorldStatic.World.Config.Player.Group1Weapon1,
+            WorldStatic.World.Config.Player.Group1Weapon2,
+            WorldStatic.World.Config.Player.Group1Weapon3,
+        },
+        {
+            WorldStatic.World.Config.Player.Group2Weapon1,
+            WorldStatic.World.Config.Player.Group2Weapon2,
+            WorldStatic.World.Config.Player.Group2Weapon3,
+        },
+        {
+            WorldStatic.World.Config.Player.Group3Weapon1,
+            WorldStatic.World.Config.Player.Group3Weapon2,
+            WorldStatic.World.Config.Player.Group3Weapon3,
+        },
+        {
+            WorldStatic.World.Config.Player.Group4Weapon1,
+            WorldStatic.World.Config.Player.Group4Weapon2,
+            WorldStatic.World.Config.Player.Group4Weapon3,
+        },
+    };
+    private int CurrentGroup = -1;
```

Comment:
Make these m_currentGroup and m_currentGroupIndex to match the rest of the code base please.

Reply:
Done.